tests/system/main.c

```diff
@@ -0,0 +1,98 @@
+#define SECTION_TEXT_MAIN __attribute__((section(".text.main")))
+#define SECTION_DATA_MAIN __attribute__((section(".data.main")))
+#define SECTION_BSS_MAIN __attribute__((section(".bss.main")))
+
+#define _printf(ptr, length)                    \
```

Comment:
Don't call it `_printf` since it has nothing to format string.
Instead, call it `printstr`.